sysutils/nut/src/opnsense/www/js/widgets/Nut.js

```diff
@@ -0,0 +1,220 @@
+// endpoint:/api/nut/diagnostics/upsstatus
+
+/*
+ * Copyright (C) 2024 Deciso B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import BaseTableWidget from "./BaseTableWidget.js";
+
+export default class Nut extends BaseTableWidget {
+  constructor() {
+    super();
+    this.statusInfo = {
+      OL: { color: "text-success", fullName: "On Line" },
+      OB: { color: "text-danger", fullName: "On Battery" },
+      LB: { color: "text-danger", fullName: "Low Battery" },
+      RB: { color: "text-warning", fullName: "Replace Battery" },
+    };
+  }
+
+  getMarkup() {
+    let $container = $("<div></div>");
+    let $nuttable = this.createTable("nut-table", {
+      headerPosition: "left",
+    });
+    $container.append($nuttable);
+    return $container;
+  }
+
+  getBatteryChargeColor(charge) {
+    if (charge == 100) return "text-success";
+    if (charge > 0) return "text-warning";
+    return "text-danger";
+  }
+
+  getUpsLoadColor(load) {
+    return load == 0 ? "text-success" : "text-warning";
+  }
+
+  async onWidgetTick() {
+    await ajaxGet("/api/nut/diagnostics/upsstatus", {}, (data, status) => {
+      let rows = [];
+      const keysOfInterest = [
+        "ups.status",
```

Comment:
Maybe these could be translated too? The other UPS widget seems to do it.

Ref1: https://github.com/opnsense/plugins/blob/64a38846959331313ac3b13a8484d267bf177c16/sysutils/apcupsd/src/opnsense/www/js/widgets/Apcupsd.js#L113-L114

Ref2: https://github.com/opnsense/plugins/blob/master/sysutils/apcupsd/src/opnsense/www/js/widgets/Metadata/Apcupsd.xml